urla05239b71c_viniciusinnovatium_mumps2java_api_tgz-pJ8-com_disclinc_netmanager_script_test_ExecTesteNewJ8/src/br/com/innovatium/mumps2java/dataaccess/ConnectionFactory.java

```diff
@@ -0,0 +1,76 @@
+package br.com.innovatium.mumps2java.dataaccess;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.Properties;
+
+import javax.naming.Context;
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+import javax.sql.DataSource;
+
+public final class ConnectionFactory {
+	private static Properties properties;
+	private static InitialContext context;
+
+	static {
+		properties = new Properties();
+		properties.put(Context.URL_PKG_PREFIXES, "org.jboss.ejb.client.naming");
+		try {
+			context = new InitialContext(properties);
+		} catch (NamingException e) {
+			throw new IllegalStateException(
+					"Fail to initialize de contexto to do lookups of the datasource.",
+					e);
+		}
+	}
+
+	private ConnectionFactory() {
+
+	}
+
+	public static Connection getConnection(ConnectionType type)
+			throws SQLException {
+
+		if (context == null) {
+			try {
+				context = new InitialContext();
+			} catch (NamingException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)